bashlex/tokenizer.py

```diff
@@ -1042,6 +1042,8 @@ def _getc(self, remove_quoted_newline=True):
 
```

Comment:
did you try following the code at bash/parse.y line 2220 to see how they deal with this?